src/main/java/com/google/api/codegen/transformer/StaticLangApiMethodTransformer.java

```diff
@@ -489,6 +489,7 @@ private void setCommonFields(
     } else {
       methodViewBuilder.hasReturnValue(method.hasReturnValue());
     }
+    methodViewBuilder.headerRequestParams(ApiCallableTransformer.getHeaderRequestParams(context));
```

Comment:
Is this ok? I'm calling into the `ApiCallableTransformer` from here, don't know if that's considered ugly or not. Anyway, I need the `headerRequestParams` field to be part of the method to do this in Go.

Reply:
That's kinda ugly. Wouldn't it be better to extract a `HeaderRequestParamsTransformer` that both transformers use?

Reply:
Sounds good, as discussed will leave to you to change in your follow up.